src/proficiency_ui.cpp

```diff
@@ -0,0 +1,375 @@
+{
+    shared_ptr_fast<ui_adaptor> current_ui = ui.lock();
+    if( !current_ui ) {
+        ui = current_ui = make_shared_fast<ui_adaptor>();
+        current_ui->on_screen_resize( [this]( ui_adaptor & cui ) {
+            init_ui_windows();
+            cui.position_from_window( catacurses::stdscr );
+        } );
+        current_ui->mark_resize();
+        current_ui->on_redraw( [this]( const ui_adaptor & ) {
+            draw_borders();
+            draw_header();
+            draw_profs();
+            draw_details();
+        } );
+    }
+    return current_ui;
+}
+
+void _prof_window::run()
+{
+    if( !u ) {
+        return;
+    }
+
+    ctxt = input_context( "PROFICIENCY_WINDOW" );
+    ctxt.register_directions();
+    ctxt.register_action( "HELP_KEYBINDINGS" );
+    ctxt.register_action( "PREV_CATEGORY" );
+    ctxt.register_action( "NEXT_CATEGORY" );
+    ctxt.register_action( "FILTER" );
+    ctxt.register_action( "QUIT" );
+
+    populate_categories();
+    shared_ptr_fast<ui_adaptor> current_ui = create_or_get_ui_adaptor();
+    init_ui_windows();
+
+    bool done = false;
+    while( !done ) {
+        ui_manager::redraw();
+        std::string action = ctxt.handle_input();
+        std::vector<display_proficiency *> &cur_set = get_current_set();
+        if( action == "UP" ) {
+            sel_prof--;
+            if( sel_prof < 0 ) {
+                sel_prof = std::max( static_cast<int>( cur_set.size() ) - 1, 0 );
+            }
+        } else if( action == "DOWN" ) {
+            sel_prof++;
+            if( sel_prof >= static_cast<int>( cur_set.size() ) ) {
+                sel_prof = 0;
+            }
+        } else if( action == "LEFT" || action == "PREV_CATEGORY" ) {
+            current_cat--;
+            if( current_cat < 0 ) {
+                current_cat = std::max( static_cast<int>( cats.size() ) - 1, 0 );
+            }
+            sel_prof = 0;
+            top_prof = 0;
+        } else if( action == "RIGHT" || action == "NEXT_CATEGORY" ) {
+            current_cat++;
+            if( current_cat >= static_cast<int>( cats.size() ) ) {
+                current_cat = 0;
+            }
+            sel_prof = 0;
+            top_prof = 0;
+        } else if( action == "FILTER" ) {
+            //~ Refers to single-character search prefixes, like p: or s:
+            std::string desc( _( "Available prefixes:" ) );
+            desc += "\n  ";
+            desc += _( "<color_c_light_cyan>l:</color>  Learned proficiencies" );
+            desc += "\n  ";
+            desc += _( "<color_c_light_cyan>i:</color>  In-progress proficiencies" );
+            desc += "\n  ";
+            desc += _( "<color_c_light_cyan>p:</color>  Returns proficiencies above the given progress.  "
+                       "ex: <color_c_yellow>p:95.00</color>" );
+            desc += "\n\n";
+            desc += _( "By default, the filter applies to proficiencies containing the given string."
```

Comment:
```suggestion
            desc += _( "By default, the filter applies to proficiencies containing the given string. "
```

Reply:
```suggestion
            desc += _( "By default, the filter applies to proficiencies containing the given string.  "
```
Cheers. I believe translations require 2 spaces after a dot, though.